Detectors/GlobalTrackingWorkflow/tpcinterpolationworkflow/include/TPCInterpolationWorkflow/TPCInterpolationSpec.h

```diff
@@ -0,0 +1,50 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_TPC_INTERPOLATION_SPEC_H
+#define O2_TPC_INTERPOLATION_SPEC_H
+
+/// @file   TPCInterpolationSpec.h
+
+#include "DataFormatsTPC/Constants.h"
+#include "SpacePoints/TrackInterpolation.h"
+#include "Framework/DataProcessorSpec.h"
+#include "Framework/Task.h"
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace tpc
+{
+class TPCInterpolationDPL : public Task
+{
+ public:
+  TPCInterpolationDPL(bool useMC, const std::vector<int>& tpcClusLanes) : mUseMC(useMC), mTPCClusLanes(tpcClusLanes) {}
+  ~TPCInterpolationDPL() override = default;
+  void init(InitContext& ic) final;
+  void run(ProcessingContext& pc) final;
+
+ private:
+  o2::tpc::TrackInterpolation mInterpolation; // track interpolation engine
+  std::vector<int> mTPCClusLanes;
+  std::array<std::vector<char>, o2::tpc::Constants::MAXSECTOR> mBufferedTPCClusters;
+
+  bool mFinished{false}; ///< finished processing flag
+  bool mUseMC{false};    ///< MC flag
```

Comment:
what would be the meaning of using MC in this workflow?

Reply:
Good question.. For Run 2 as far as I understood the maps were created only from real data and not from MC. And in the reconstruction of MC data (for which the TPC cluster positions were distorted according to the inverse of the correction maps and some randomness) the same correction maps which were obtained in real data are used.
If the procedure in Run 3 is similar then MC information is not really needed for the interpolation, right? I mean the matching performance is checked before in the global reconstruction workflow and this does not need to be repeated here.

Reply:
The only usage of the MC info I can imaging is to check by how much the filtering procedure before interpolation rejects the fake matches. So, at the moment I would allow receiving the MClabels input but would not propagate anything MC-related to the output.

Reply:
@martenole is this ready to be tested/merged ?